### PR TITLE
Make it work!

### PR DESCRIPTION
**Pull request by cadecairos**
Work in Progress, here so people can track things.
- [x] implement build system
- [x] build login ux
- [x] build navbar ux
- [x] build one-up view

### CODE REVIEW

**Comment by thisandagain** [comment]
:smile:

**Comment by alicoding** [comment]
- [ ] When removing a make the frame should be reloaded

File: firehose/components/Firehose.jsx [hunk 1]
```diff
@@ -1,43 +1,86 @@
...
+var FirehoseEditor = require("./views/FirehoseEditor.jsx");
+var FirehoseActions = require("../actions/FirehoseActions");
 
 var Firehose = React.createClass({
   getInitialState: function() {
     return {
       loggedIn: false,
-      isAdmin: false
+      isAdmin: false,
+      csrfToken: document.querySelector("meta[name='csrf-token']").getAttribute("content"),
```
Comment (alicoding) [inline firehose/components/Firehose.jsx]: Move it to `componentDidMount()`

File: firehose/components/header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+var React = require("react");
+var WebmakerLoginUX = require("./WebmakerLoginUX.jsx");
+var ViewToggle = require("./ViewToggle.jsx");
+
+var Header = React.createClass({
+  getInitialState: function() {
+    return {
+      loggedIn: false,
+      isAdmin: false,
+      viewState: this.props.viewState
```
Comment (alicoding) [inline firehose/components/header/Header.jsx]: Delete this line and use `this.props.viewState` instead.

File: firehose/components/header/ViewToggle.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+var React = require("react");
+
+var ViewToggle = React.createClass({
+  getInitialState: function() {
+    return {
+      viewState: this.props.viewState
```
Comment (alicoding) [inline firehose/components/header/ViewToggle.jsx]: Here too

File: firehose/components/shared/Navigate.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+var React = require("react");
+var FirehoseActions = require("../../actions/FirehoseActions");
+var FirehoseConstants = require("../../actions/FirehoseConstants");
+
+var Navigate = React.createClass({
+  getInitialState: function() {
+    var delta = +this.props.delta;
```
Comment (alicoding) [inline firehose/components/shared/Navigate.jsx]: Here too

File: firehose/components/MakePreview.jsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+var React = require("react");
```
Comment (alicoding) [inline firehose/components/MakePreview.jsx]: REMOVE THIS FILE!!!

**Comment by cadecairos** [comment]
> When removing a make the frame should be reloaded

not sure how to trigger this - when I delete the last make, I get the "no makes" message.

**Comment by alicoding** [comment]
great patch! r+

**Comment by alicoding** [comment]
GOOD ONE DUDE! :D

**Pull request merged**